src/components/__tests__/SubThought.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import App from '../../App'
+import { getThoughtsRanked } from '../../util.js'
+
+// eslint-disable-next-line fp/no-let
+let wrapper = null
+
+beforeAll(async () => {
+  wrapper = await window.createTestApp(<App />)
+})
+
+afterAll(() => {
+  window.cleanupTestApp()
+  wrapper = null
+})
+
+it('create and edit subthought', async () => {
+  const thoughts = wrapper.find('div.transformContain div ul')
+  const thought = thoughts.find(
+    'li.leaf div.thought-container div.thought div.editable',
+  )
+  expect(thought.text()).toBe('')
+  const keyboardResponder = wrapper.find('div#keyboard')
+  await thought.simulate('change', { target: { value: 'c' } })
+  await keyboardResponder.simulate('keydown', { key: 'Enter', ctrlKey: true })
+  jest.runAllTimers()
+  await thought.update()
+  expect(thought.text()).toBe('c')
+  // eslint-disable-next-line fp/no-let
+  let subthoughtsData = getThoughtsRanked(['c'])
+  // eslint-disable-next-line fp/no-let
+  let subthoughtData = subthoughtsData[0]
+  expect(subthoughtData.value).toBe('')
+  await wrapper.update()
+  const subthought = wrapper.find(
+    'ul.distance-from-cursor-0 li.leaf div.thought-container div.thought div.editable',
```

Comment:
In general, we want the minimal, semantic selector that identifies the relevant element for the test. This will be more resilient to markup changes.

Reply:
I tried. I even tried selectors with components. If you do find selector that is more general and works, please do tell!

Reply:
Okay, thanks for trying!